tests/components/bthome/test_sensor.py

```diff
@@ -393,47 +230,10 @@ def _async_register_callback(_hass, _callback, _matcher, _mode):
         saved_callback = _callback
         return lambda: None
 
-    with patch(
-        "homeassistant.components.bluetooth.update_coordinator.async_register_callback",
-        _async_register_callback,
-    ):
-        assert await hass.config_entries.async_setup(entry.entry_id)
-        await hass.async_block_till_done()
-
-    assert len(hass.states.async_all()) == 0
-    saved_callback(
-        make_advertisement("A4:C1:38:8D:18:B2", b"\x03\x12\xe2\x04"),
-        BluetoothChange.ADVERTISEMENT,
-    )
-    await hass.async_block_till_done()
-    assert len(hass.states.async_all()) == 1
-
-    sensor = hass.states.get("sensor.test_device_18b2_carbon_dioxide")
-    sensor_attr = sensor.attributes
-    assert sensor.state == "1250"
-
-    assert sensor_attr[ATTR_FRIENDLY_NAME] == "Test Device 18B2 Carbon Dioxide"
-    assert sensor_attr[ATTR_UNIT_OF_MEASUREMENT] == "ppm"
-    assert sensor_attr[ATTR_STATE_CLASS] == "measurement"
-
-    assert await hass.config_entries.async_unload(entry.entry_id)
-    await hass.async_block_till_done()
-
-
-async def test_bthome_voc(hass):
-    """Make sure that VOC sensors are correctly mapped."""
-    entry = MockConfigEntry(
-        domain=DOMAIN,
-        unique_id="A4:C1:38:8D:18:B2",
-    )
-    entry.add_to_hass(hass)
-
-    saved_callback = None
-
-    def _async_register_callback(_hass, _callback, _matcher, _mode):
-        nonlocal saved_callback
-        saved_callback = _callback
-        return lambda: None
+    if entry.data["bindkey"]:
+        advertisement = make_encrypted_advertisement(mac_address, service_data)
```

Comment:
Let's pass the function to create the advertisement as a parameter instead. We try to avoid conditions in tests.

Reply:
Thx, moved the function to a parameter